src/Compilers/Core/Portable/Operations/IReDimOperation.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Operations
+{
+    /// <summary>
+    /// Represents the ReDim operation to re-allocate storage space for array variables.
+    /// <para>
+    /// Current usage:
+    ///  (1) VB ReDim statement.
+    /// </para>
+    /// </summary>
+    /// <remarks>
+    /// This interface is reserved for implementation by its associated APIs. We reserve the right to
+    /// change it in the future.
+    /// </remarks>
+    public interface IReDimOperation : IOperation
```

Comment:
ReDim seems to be a VB specific term. Is there any precedent for even using "dim" somewhere in IOperation? If not, should this be something language neutral such as IArrayResizeOperation?

Reply:
Thanks, that sounds like a good suggestion. We will discuss it at next week's IOperation design meeting.

Reply:
We decided on the using the "ReDim" term in the design meeting, given that the operation has very VB specific semantics (such as you can ReDim a variable which has been statically typed as an `Object` to an array of any type). This also falls in line with other such language specific nodes such as IWithOperation, IStopOperation, etc.

---
In reply to: [221343469](https://github.com/dotnet/roslyn/pull/30217#discussion_r221343469) [](ancestors = 221343469)